packages/firestore/src/typings/firestore.d.ts

```diff
@@ -0,0 +1,1026 @@
+/**
```

Comment:
So it seems our type definitions are now duplicated in 3 places (here and in app's firebase.d.ts and app.d.ts). Is there any way to avoid the redundancy?  I really don't want to update 3 files in 2 different packages every time we modify an API.

Reply:
Agreed, I am looking into making it so the wrapper package (i.e. `firebase`) leverages the generated typings from each of the downstream packages. There are some limitations that make that difficult and is going to require some refactoring to make it possible. 

I opted to not refactor the typings too much initially in favor of later cleanup.